rootfs/etc/nginx/lua/balancer/sticky.lua

```diff
@@ -99,4 +96,21 @@ function _M.balance(self)
   return self.instance:find(key)
 end
 
+function _M.sync(self, backend)
+  balancer_resty.sync(self, backend)
+
+  -- Reload the balancer if any of the annotations have changed.
+  local changed = not util.deep_compare(
+    self.cookie_session_affinity,
+    backend.sessionAffinityConfig.cookieSessionAffinity
+  )
+  if not changed then
+    return
+  end
+
+  for key, val in pairs(self:new(backend)) do
```

Comment:
This is not good - what you wanna do here is to update only the things that might change when `self.cookie_session_affinity` changes. But right now you are also creating a new resty balancer instance and updating the instance of self with new one unnecessarily.

All you need to do here is to update `cookie_session_affinity` and `digest_func`.

Reply:
fixed